connector/src/rest/connector/libs/nxos/implementation.py

```diff
@@ -103,17 +103,33 @@ def connect(self, timeout=30):
             >>> device = testbed.devices['asr22']
             >>> device.connect(alias='rest', via='rest')
         '''
+        protocol = 'https'
 
         if self.connected:
             return
 
-        ip = self.connection_info['ip'].exploded
-
-        if 'port' in self.connection_info:
-            port = self.connection_info['port']
-            self.url = 'http://{ip}:{port}'.format(ip=ip, port=port)
+        # support sshtunnel
+        if 'sshtunnel' in self.connection_info:
+            from unicon.sshutils import sshtunnel
```

Comment:
import usually always goes to the top,  why this one is special?

and because rest is outside of unicon, i would do

try except around it, and if not available set a flag; if flag is not set, ssh tunel is not available

Reply:
added `try/except` but keep having it at same place. but I think we don't want to see exception when user is not using `sshtunnel`. so better to have `import` with try/except after checking if `sshtunnel` is specified in yaml.